Refactor with Defensive Patterns

rsandor opened the pull request:
- Companies delegate now uses companies intercom client instead of full orion base class
- Moved `canUseIntercom` out into a utility library
- Added defensive programming in `_wrap` method to avoid unexpected throws
- Refactored and simplified tests
## Reviewers
- [x] @podviaznikov 
- [x] @Myztiq

Commit:
Refactor with Defensive Patterns

- Companies delegate now uses companies intercom client instead of full orion base class
- Moved `canUseIntercom` out into a utility library
- Added defensive programming in `_wrap` method to avoid unexpected throws
- Refactored and simplified tests

podviaznikov commented on lib/companies.js:
```diff
@@ -1,57 +1,75 @@
+
+    return clientMethod.apply(
+      clientMethod,
+      Array.prototype.slice.call(arguments, 1)
```
We are skipping first argument because it's method name?

podviaznikov replied:
just add comment

Myztiq commented on lib/companies.js:
```diff
@@ -1,57 +1,75 @@
+   *   returns results of call to intercom client
    * @private
    */
   _wrap (method) {
```
Long term - We are probably going to want this in a base class that's inherited.

rsandor replied:
Indeed, no need to pre-abstract imho.

Myztiq commented on lib/companies.js:
```diff
@@ -1,57 +1,75 @@
+
+    const clientMethod = this.client[method]
+    if (!isFunction(clientMethod)) {
+      return Promise.reject(
```
I'd prefer we use "Throw" and use Promise.method (http://bluebirdjs.com/docs/api/promise.method.html)

Commit:
Minor fixes based on review

- Added comment about skipped argument in `_wrap`
- Use a better promise style for `_wrap`

podviaznikov commented:
this looks good

podviaznikov commented:
@rsandor can you also please create travis integration if that is not a big problem for you?

rsandor commented:
@podviaznikov - I don't really have the time for that, can we find someone to do it in a separate PR?

podviaznikov commented:
sure, np
